unstructured/ingest/v2/README.md

```diff
@@ -0,0 +1,119 @@
+- [ ] Update ingest documentation [here](https://github.com/Unstructured-IO/docs)
+- [ ] For team members that are developing in the original repository:
+  - [ ] If there are secret variables created for the connector tests, make sure to:
+    - [ ] add the secrets into Github (contact someone with access)
+    - [ ] include the secret variables in [`ci.yml`](https://github.com/Unstructured-IO/unstructured/blob/main/.github/workflows/ci.yml) and [`ingest-test-fixtures-update-pr.yml`](https://github.com/Unstructured-IO/unstructured/blob/main/.github/workflows/ingest-test-fixtures-update-pr.yml)
+    - [ ] add a make install line in the workflow configurations to be able to provide the workflow machine with the required dependencies on the connector while testing
+    - [ ] Whenever necessary, use the [ingest update test fixtures](https://github.com/Unstructured-IO/unstructured/actions/workflows/ingest-test-fixtures-update-pr.yml) workflow to update the test fixtures.
+
+## Design References
+
+The ingest flow is similar to an ETL pipeline that gets defined at runtime based on user input:
+
+![unstructured ingest cli diagram](assets/pipeline.png)
+
+
+
+### Steps
+* `Indexer`: This is responsible for reaching out to the source location and pulling in metadata for each document that will need to be downloaded and processed
+* `Downloader`: Using the information generated from the indexer, download the content as files on the local file system for processing. This may require manipulation of the data to prep it for partitioning.
+* `Uncompressor`: If enabled, this will look for any supported compressed files (tar and zip are currently supported) and expands those.
+* `Partitioner`: Generated the structured enriched content from the local files that have been pulled down. Both local and api-based partitioning is supported, with api-based partitioning set to run async while local set to run via multiprocessing.
+* `Chunker`: Optionally chunk the partitioned content. Can also be run locally or via the api, with async/multiprocessing set in the same approach as the partitioner.
+* `Embedder`: Create vector embeddings for each element in the structured output. Most of these are via an API call (i.e. AWS Bedrock) and therefor run async but can also use a local huggingface model which will run via multiprocessing.
+* `Stager`: This is an optional step that won't apply for most pipelines. If the data needs to be modified from the existing structure to better support the upload, such as converting it to a csv for tabular-based destinations (sql dbs).
+* `Uploader`: Write the local content to the destination. If none if provided, the local one will be used which writes the final result to a location on the local filesystem. If batch uploads are needed, this will run in a single process with access to all docs. If batch is not supported, all docs can be uploaded concurrently using the async approach.
+
+
+
+### Parallel Execution/Concurrency
```

Comment:
The above is a good birds eye view; I think would be a really helpful addition is to diagram and explain the entire journey in depth. i.e. trace granularly (like function level) incoming request -> constructing command -> building pipeline -> executing pipeline so that a developer can drop into code an understand exactly where it fits in and how to debug.